src/input/mod.rs

```diff
@@ -972,17 +980,19 @@ impl State {
                         std::mem::drop(shell);
                     };
 
-                    let ptr = seat.get_pointer().unwrap();
-                    ptr.button(
-                        self,
-                        &ButtonEvent {
-                            button,
-                            state: event.state(),
-                            serial,
-                            time: event.time_msec(),
-                        },
-                    );
-                    ptr.frame(self);
+                    if pass_event {
```

Comment:
This isn't entirely correct, as this would still send the `ButtonReleased`-event without the client seeing a `ButtonPressed`-event previously.

You might want to take inspiration from `surpressed_keys` for keyboard-input. This will need a similar solution for mouse buttons.

Reply:
I've been trying to fix this for a while and I think I might have taken the wrong approach, contrary to the keyboard `input` where we can intercept the event from the client, we must always send the `button` event because it will be sent to the `PointerGrab`, otherwise it will get stuck.

My idea is that the interception code gets moved to the `MoveGrab` and `ResizeForkGrab` where a new boolean field gets added for whether they are grabs resulting from a Super + {Left, Right} Click or not, And if they are the `button` handler doesn't pass the event to the client.

What do you think @Drakulix ?

Reply:
I've actually implemented the suppress functionality in the grabs and ended up needing to do both the current approach of suppressing pressed events in `process_input` and release events in the grabs, because the grab is only added in the next idle cycle.

However this still seems to be not enough because in case a grab is not set there exists nothing to suppress the release event. So it seems like something like `suppressed_keys` is also needed.

Reply:
I've finished adding code to suppress buttons that do not trigger a grab.

Honestly the code is not ideal, there's a lot of argument drilling but I don't know how to architect it better.